qCC/ccCommandLineParser.cpp

```diff
@@ -390,6 +390,200 @@ QString ccCommandLineParser::exportEntity(	CLEntityDesc& entityDesc,
 	return (result != CC_FERR_NO_ERROR ? QString("Failed to save result in file '%1'").arg(outputFilename) : QString());
 }
 
+bool ccCommandLineParser::selectClouds(bool selectAll, bool reverse, bool selectFirst, bool selectLast, bool selectRegex, int firstNr, int lastNr, QRegExp regex)
+{
+	if (selectRegex && !regex.isValid())
+	{
+		return error(QObject::tr("Regex string invalid: %1").arg(regex.errorString()));
+	}
+
+	//store everthyng in the rest vector
+	while (!m_clouds.empty())
+	{
+		m_clouds_rest.push_back(m_clouds.front());
+		m_clouds.erase(m_clouds.begin());
+	}
+
+	//sort the rest clouds by uniqueID (it will sort the clouds as it were loaded/created)
+	std::sort(m_clouds_rest.begin(), m_clouds_rest.end(), [](CLCloudDesc a, CLCloudDesc b) { return (a.pc->getUniqueID() < b.pc->getUniqueID()); });
+
+	//put elements to the front facing vector
+	int index = 0;
+	size_t lastIndex = m_clouds_rest.size() - 1;
```

Comment:
you have to test before that m_unselectedClouds is not empty (otherwise it will cause an underflow and more issues later...)